sdk/core/Azure.Core/src/Internal/ModelWriter.SequenceBuilder.cs

```diff
@@ -0,0 +1,166 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Buffers;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure.Core.Internal
+{
+    internal sealed partial class ModelWriter : IDisposable
+    {
+        private sealed class SequenceBuilder : IBufferWriter<byte>, IDisposable
+        {
+            private struct Buffer
+            {
+                public byte[] Array;
```

Comment:
Can we make `Array` readonly?

Reply:
it is updated after initialization since some buffers are "unused"